loader_hub/web/beautiful_soup_web/base.py

```diff
@@ -15,16 +15,67 @@ def _substack_reader(soup: Any) -> Tuple[str, Dict[str, Any]]:
     text = soup.select_one("div.available-content").getText()
     return text, extra_info
 
+def _readthedocs_reader(soup: Any,url: str) -> Tuple[str, Dict[str, Any]]:
+    """Extract text from a ReadTheDocs documentation site
 
-def _readthedocs_reader(soup: Any) -> Tuple[str, Dict[str, Any]]:
-    """Extract text from a ReadTheDocs documentation site"""
-    text = soup.find_all("main", {"id": "main-content"})[0].get_text()
-    return "\n".join([t for t in text.split("\n") if t]), {}
+    """
+    links = soup.find_all('a', {'class': 'reference internal'})
+    rtd_links = []
+    
+    for link in links:
+        rtd_links.append(link['href'])   
+    for i in range(len(rtd_links)):
+        if not rtd_links[i].startswith("http"):
+            rtd_links[i] = url + rtd_links[i]
+    texts = []
+    broken_docs = []
+    
+    for doc_link in rtd_links:
+        import requests
```

Comment:
These can be pulled out of the for loop and just put at the top of this function